DateTimeCompareValidator.php

```diff
@@ -152,12 +156,13 @@ protected function validateValue($value)
      * @param string $operator the comparison operator
      * @param mixed $value the value being compared
      * @param mixed $compareValue another value being compared
+     * @params string $format the date format to use
      * @return boolean whether the comparison using the specified operator is true.
      */
-    protected function compareValues($operator, $value, $compareValue)
+    protected function compareValues($operator, $value, $compareValue,$format)
     {
-    	$dateValue = new \DateTime($value);
-    	$dateCompareValue = new \DateTime($compareValue);
+    	$dateValue = \DateTime::createFromFormat($format, $value);
+        $dateCompareValue = \DateTime::createFromFormat($format, $compareValue);
```

Comment:
Remove the extra space before ```$dateCompareValue```. Also, use the old DateTime constructor if no ```$format``` is given.

```
$dateValue = isset($format) ? \DateTime::createFromFormat($format, $value) : new \DateTime($value);
$dateCompareValue = isset($format) ? \DateTime::createFromFormat($format, $compareValue) : new \DateTime($compareValue);
```